setup-le.sh

```diff
@@ -1,14 +1,29 @@
 #!/usr/bin/bash
 set -o nounset -o errexit
 
+sed -i "s/server.example.test/`hostname -f`/g"
+
 WORKDIR=$(dirname "$(realpath $0)")
 
 dnf install letsencrypt -y
 
-ipa-cacert-manage install "$WORKDIR/ca/DSTRootCAX3.pem" -n DSTRootCAX3 -t C,,
-ipa-certupdate -v
+if [ ! -d /etc/ssl/`hostname -f` ]
+then
+  mkdir -p /etc/ssl/`hostname -f`
+fi
 
-ipa-cacert-manage install "$WORKDIR/ca/LetsEncryptAuthorityX3.pem" -n letsencryptx3 -t C,,
-ipa-certupdate -v
+wget -O /etc/ssl/`hostname -f`/isrgrootx1.pem  https://letsencrypt.org/certs/isrgrootx1.pem
+wget -O /etc/ssl/`hostname -f`/isrg-root-x2.pem https://letsencrypt.org/certs/isrg-root-x2.pem
+wget -O /etc/ssl/`hostname -f`/lets-encrypt-r3.pem https://letsencrypt.org/certs/lets-encrypt-r3.pem
+wget -O /etc/ssl/`hostname -f`/lets-encrypt-e1.pem https://letsencrypt.org/certs/lets-encrypt-e1.pem
+wget -O /etc/ssl/`hostname -f`/lets-encrypt-r4.pem https://letsencrypt.org/certs/lets-encrypt-r4.pem
+wget -O /etc/ssl/`hostname -f`/lets-encrypt-e2.pem https://letsencrypt.org/certs/lets-encrypt-e2.pem
+ipa-cacert-manage install /etc/ssl/`hostname -f`/isrgrootx1.pem
+ipa-cacert-manage install /etc/ssl/`hostname -f`/isrg-root-x2.pem
+ipa-cacert-manage install /etc/ssl/`hostname -f`/lets-encrypt-r3.pem
+ipa-cacert-manage install /etc/ssl/`hostname -f`/lets-encrypt-e1.pem
+ipa-cacert-manage install /etc/ssl/`hostname -f`/lets-encrypt-r4.pem
+ipa-cacert-manage install /etc/ssl/`hostname -f`/lets-encrypt-e2.pem
+ipa-certupdate
```

Comment:
Rather than calling hostname -f so many times why not use a variable?

I realize this is a pretty basic script in general but there is no error handling at all and the wget could fail (assuming wget is installed) or the cacert-manage could fail and it would charge on producing a fair bit of failed output.

Reply:
> Rather than calling hostname -f so many times why not use a variable?

It was just an artifact of laziness and simplicity.  This can be changed easy enough, however.

> I realize this is a pretty basic script in general but there is no error handling at all and the wget could fail (assuming wget is installed) or the cacert-manage could fail and it would charge on producing a fair bit of failed output.

Fair point. I can put in some logic to at least look for wget/curl and use whatever is installed.

Reply:
Ok.  Sorry for all the commits, I didn't pull locally to my pc so it just got very ugly quickly.

The new commits include the changes discussed here as well as handling to detect if curl/wget was installed and run the download with whichever is installed.

Reply:
This looks good. Some of the certificate removal patches didn't apply cleanly for some reason. Confusing since github seems to think they're fine.

Can you squash all the commits down? Maybe one for "remove unneeded local copy of LE cert chain" and one for the setup-le.sh script? Also, can you add a reference to the originating ticket, https://github.com/freeipa/freeipa-letsencrypt/issues/25

e.g. A commit message something like

```
Update setup-le.sh to dynamically retrieve LE CA chain

Detect wget or curl availability and retrieve the set of current certs
from LE directly and import them into IPA.

Use the current hostname to configure the CSR for the LE
certificate request.    

Fixes: https://github.com/freeipa/freeipa-letsencrypt/issues/25
Signed-off-by: Your Name <something@somewhere>
```

Reply:
I will work on this today/tomorrow and will let you know when done.  Thank you!

Reply:
Ok this is done!

Reply:
@rcritten  - Just wondering if this is appropriate to commit?  Just want to make sure that you aren't waiting on anything from me.